src/formatDate.js

```diff
@@ -7,8 +7,70 @@
  *
  * @returns {string}
  */
+
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const newDateArr = [];
+  let month = '';
+  let day = '';
+  let year = '';
+
+  const separator = fromFormat[fromFormat.length - 1];
+  const newSeparator = toFormat[toFormat.length - 1];
+  const dateArr = date.split(separator);
+
+  const requiredYearFormat = toFormat.find(
+    (element) => element === 'YYYY' || element === 'YY',
+  );
+
+  const changeYearFormat = () => {
+    if (requiredYearFormat.length === 4 && year.length === 4) {
+      return year;
+    }
+
+    if (requiredYearFormat.length === 4 && year < 30) {
+      return `20${year}`;
+    }
+
+    if (requiredYearFormat.length === 4 && year >= 30) {
+      return `19${year}`;
+    }
+
+    if (requiredYearFormat.length === 2) {
+      return year.slice(-2);
+    }
+  };
+
+  for (let i = 0; i < fromFormat.length; i++) {
+    if (fromFormat[i] === 'YYYY' || fromFormat[i] === 'YY') {
+      year = dateArr[i];
+    }
+
+    if (fromFormat[i] === 'DD') {
+      day = dateArr[i];
+    }
+
+    if (fromFormat[i] === 'MM') {
+      month = dateArr[i];
+    }
+  }
+
+  year = changeYearFormat();
+
+  for (let j = 0; j < toFormat.length; j++) {
```

Comment:
Similar to the previous comment, the for-loop used to construct the new date string based on 'toFormat' could also benefit from using a higher-order function like 'map' to transform the 'toFormat' array directly into the 'newDateArr'.